### ISSUE TITLE
How to compile for x86 Unity Platform Build?

### ISSUE THREAD

**Issue by ShumWengSang** [Joelx/BleWinrtDll-Unity-Demo#2]
Hi, so I've converted this sample to work with all the HoloLens 1 MRTK stuff. In Unity Editor (which is x64), it works fine. However, my target platform is x86 UWP HoloLens. When I compiled the BleWinrtDLL in x86 and imported it to Unity, Unity would complain that it wasn't an x64 build. But if I use the x64 version, deploying to HoloLens would crash.

I guess this is more of a Unity question, but how would I deploy this correctly in Unity?

**Comment by marianylund** [comment]
Hey, I got the same error and to solve it I:

1. Rebuilt Dll as ARM64 Architecture, following [this ](https://developer.qualcomm.com/sites/default/files/docs/adreno-gpu/developer-guide/tutorials/windows/get_started_wos.html) guide: 

- In Visual Studio, click Build > Open Configuration Manager.
- In the Active Solution Platform list, select <New…>.
- In the New Solution Platform dialog box, select the following values:
– Type or select the new platform list: Arm64 – Copy settings from list: x64

2. In Unity I added Dll and clicked on it. In the inspector you should be able to change its platform settings. There I changed the values:
- SDK: UWP
- CPU: ARM64
- Don't process: Checked

Alas, all of this changed my error message to "Failed to load BleWinrtDll.dll because one or more of its dependencies could not be loaded." So I am off searching for more solutions. But I hope this helps!

**Comment by ShumWengSang** [comment]
@marianylund Thanks a lot. I managed to work around the Unity issue by doing using a preprocessor to detect if the build was an editor or a standalone build. If editor, I'd use the x64 and if standalone I'd use the x86 dll.

 The other issue would be the x86 dll causing issues in HoloLens. This was fixed by following the steps described [here](https://docs.microsoft.com/en-us/cpp/porting/how-to-use-existing-cpp-code-in-a-universal-windows-platform-app?view=msvc-170#to-port-a-native-dll-to-the-uwp-without-creating-a-new-project).
 
 I hope this helps!
